u2-routing-controllers/_services/fakedb.js

```diff
@@ -0,0 +1,59 @@
+const users = {
+  0: {
+    name: 'Charles',
+    email: 'charlesa@nu.ac.th',
+    skills: ['talking']
+  },
+  1: {
+    name: 'Mook',
+    email: 'areeyac62@nu.ac.th',
+
+    skills: ['js', 'node', 'expressjs', 'terminal']
+  },
+  2: {
+    name: 'Joker',
+    email: 'kanokpolt62@nu.ac.th',
+    skills: ['js', 'node', 'expressjs', 'terminal']
+  },
+  3: {
+    name: 'Oat',
+    email: 'nuttapolt62@nu.ac.th',
+
+    skills: ['js', 'node', 'expressjs', 'terminal']
+  },
+  4: {
+    name: 'Chinjung',
+    email: 'parinyak62@nu.ac.th',
+    skills: ['js', 'node', 'expressjs', 'terminal']
+  }
+}
+const findUser = (id) => {
+  return users[id]
+}
+
+const findUserByUsername = (username) => {
+  return Object.values(users).find(
+    user => username === user.email.split('@')[0])
+}
+
+const findUserBySearch = (search) => {
+  const lowerSearch = search.toLowerCase()
+  return Object.values(users).find(
+    user => user.name.toLowerCase().includes(lowerSearch))
+}
+
+const products = {
+  lenovo: {},
+  asus: {}
+}
+
+const findProducts = (q) => {
+  return products[q]
+}
```

Comment:
ทำได้ดี! นี่เป็นมากกว่าที่ฉันคาดไว้ ฉันแค่คาดหวังให้คุณได้รับข้อความค้นหาจาก URL และแสดงมัน ฉันไม่ได้คาดหวังว่าคุณจะได้รับข้อมูลที่เกี่ยวข้องเช่นกัน นี่เป็นความผิดของฉันที่ไม่ชัดเจน แต่คุณก็ประสบความสำเร็จอยู่ดี :) / Well done! This is more than I expected. I was only expecting you to get the search term from the URL and just display it, not to also look up related data. This is my fault for being unclear. But you succeeded in doing more than I hoped :)

รหัสใน `fakedb.js` อยู่นอกเหนือขอบเขตของหลักสูตร เราจะศึกษาฐานข้อมูลจริงใน U4 - รหัสนั้นจะมีประโยชน์มากกว่าสำหรับคุณ! / The code in `fakedb.js` is beyond the scope of the course. We will study real databases in U4 -- that code will be more useful for you!